rfd/0081/README.md

```diff
@@ -4,3 +4,58 @@ state: predraft
 ---
 
 # RFD 81 servicesHealth implementation
+
+The Mariposa (RFD36) health check agent will be responsible for monitoring the health of user services. The Health Check agent process(es) will read user defined health check definitions from the Project service, and coordinate which health check process will be responsible for running any given definition. The Project service is read at agent start time, then polled on an interval. There's an endpoint exposed by the agent that will trigger a poll as well.
```

Comment:
You raise some great points. I hadn't considered the needed interactions between Convergence and the health check agent(s). I'm having a difficult time imagining how the Convergence service will "spawn" the agent in the proper sec-nat zone, but surely it's responsible for managing the health check agent once it's running (adding/removing health checks as services start/stop as you mentioned). The convergence service will need a copy of the manifest anyway to do it's work; so there's no reason we couldn't go with that architecture, and have the health check agents communicate exclusively with the convergence service.